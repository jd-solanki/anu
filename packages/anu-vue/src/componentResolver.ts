import type { ComponentResolver } from 'unplugin-vue-components'

export function AnuComponentResolver(): ComponentResolver {
  return {
    type: 'component',
    resolve: (name: string) => {
      if (name.match(/^A[A-Z]/))
        return { name, from: 'anu-vue' }
    },
  }
}
