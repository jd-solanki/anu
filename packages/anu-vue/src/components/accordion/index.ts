import AAccordion from './AAccordion.vue'

// export * from './meta'
export { AAccordion }

export type AAccordion = InstanceType<typeof AAccordion>