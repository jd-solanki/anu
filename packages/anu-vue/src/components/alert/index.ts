import AAlert from './AAlert.vue'

export * from './meta'
export { AAlert }

// eslint-disable-next-line @typescript-eslint/no-redeclare
export type AAlert = InstanceType<typeof AAlert>
