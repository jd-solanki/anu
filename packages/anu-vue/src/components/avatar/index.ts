import AAvatar from './AAvatar.vue'

export * from './meta'
export { AAvatar }

// eslint-disable-next-line @typescript-eslint/no-redeclare
export type AAvatar = InstanceType<typeof AAvatar>
