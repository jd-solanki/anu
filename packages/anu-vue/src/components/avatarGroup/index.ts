export { default as AAvatarGroup } from './AavatarGroup.vue'
