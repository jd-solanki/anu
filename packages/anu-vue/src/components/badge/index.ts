import ABadge from './ABadge.vue'

export * from './meta'
export { ABadge }

// eslint-disable-next-line @typescript-eslint/no-redeclare
export type ABadge = InstanceType<typeof ABadge>
