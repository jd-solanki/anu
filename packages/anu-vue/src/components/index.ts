export * from '@/transitions'
export { AAlert } from './alert'
export { AAvatar } from './avatar'
export { ABadge } from './badge'
export { ABaseInput } from './base-input'
export { ABtn } from './btn'
export { ACard } from './card'
export { ACheckbox } from './checkbox'
export { AChip } from './chip'
export { AConfig } from './config'
export { ADataTable } from './data-table'
export type { ADataTableItemsFunction, ADataTableItemsFunctionParams, ADataTableProps } from './data-table'
export { ADialog } from './dialog'
export { ADrawer } from './drawer'
export { AFloating } from './floating'
export type { AFloatingMiddlewareFunc } from './floating'
export { AIcon } from './icon'
export { AInput } from './input'
export { AList } from './list'
export { AListItem } from './list-item'
export { ALoader } from './loader'
export { ALoadingIcon } from './loading-icon'
export { AMenu } from './menu'
export { ARadio } from './radio'
export { ARating } from './rating'
export { ASelect } from './select'
export { ASpinner } from './spinner'
export { ASwitch } from './switch'
export { ATab } from './tab'
export { ATable } from './table'
export type { ATablePropColumn, ATableProps } from './table'
export { ATabs } from './tabs'
export { ATextarea } from './textarea'
export { ATooltip } from './tooltip'
export { ATypography } from './typography'
export { AView } from './view'
export { AViews } from './views'
