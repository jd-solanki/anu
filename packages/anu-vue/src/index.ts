import '@/scss/index.scss'

export { AnuComponentResolver } from './componentResolver'
export * from './components'
export * as components from './components'
export * from './composables'
export * as composables from './composables'
export { plugin as anu } from './plugin'
export type { PluginOptions, ThemeColors, ThemeOptions as ThemeOption } from './plugin'
export { defaultThemeColors, presetAnu } from './preset'
export { presetIconExtraProperties } from './preset/icons'
export * from './symbols'
