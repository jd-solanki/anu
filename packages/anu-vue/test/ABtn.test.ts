import { shallowMount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import { ABtn } from '../src/components'

describe('Testing ABtn', () => {
  it('can mount component', () => {
    const wrapper = shallowMount(ABtn, { props: {}, slots: { default: 'Click me' } })

    expect(wrapper.find('button').exists()).toBe(true)
    expect(wrapper.find('button').text()).toBe('Click me')

    expect(wrapper.classes()).toContain('a-btn')

    // In browser mode, this is invalid
    // const btnStyles = window.getComputedStyle(wrapper.find('button').element)._values
    // expect(btnStyles['--a-layer-c']).toBe('var(--a-primary)')

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('has proper classes', () => {
    const wrapper = shallowMount(ABtn, { props: {}, slots: { default: 'Click me' } })
    expect(wrapper.classes()).toMatchSnapshot()
  })

  it('handles click event', async () => {
    const wrapper = shallowMount(ABtn, { props: {}, slots: { default: 'Click me' } })

    await wrapper.trigger('click')
    expect(wrapper.emitted('click')).toBeDefined()
  })

  // Invalid in browser
  // it('color prop changes button color', () => {
  //   const wrapper = shallowMount(ABtn, { props: { color: 'warning' }, slots: { default: 'Click me' } })

  //   const btnStyles = window.getComputedStyle(wrapper.find('button').element)._values

  //   expect(btnStyles['--a-layer-c']).toBe('var(--a-warning)')
  // })

  it('icon prop adds an icon', () => {
    const wrapper = shallowMount(ABtn, { props: { icon: 'i-bx-heart' }, slots: { default: 'Click me' } })

    expect(wrapper.findAll('button i')[0].classes()).toContain('i-bx-heart')
  })

  it('appendIcon prop adds an icon after the text', () => {
    const wrapper = shallowMount(ABtn, { props: { icon: 'i-bx-heart', appendIcon: 'i-bx-star' }, slots: { default: 'Click me' } })

    expect(wrapper.findAll('button i')[0].classes()).toContain('i-bx-heart')
    expect(wrapper.findAll('button i')[1].classes()).toContain('i-bx-star')
  })

  it('iconOnly prop applies a-btn-icon-only style', () => {
    const wrapper = shallowMount(ABtn, { props: { icon: 'i-bx-heart', iconOnly: true }, slots: { default: 'Click me' } })

    expect(wrapper.findAll('button i')[0].classes()).toContain('i-bx-heart')
    expect(wrapper.classes()).not.toContain('a-btn')
    expect(wrapper.classes()).toContain('a-btn-icon-only')
  })

  it('disabled prop disables cursor, decreases opacity and set tabindex to -1', async () => {
    const wrapper = shallowMount(ABtn, { props: { disabled: true }, slots: { default: 'Click me' } })

    expect(wrapper.classes()).toContain('opacity-50')
    expect(wrapper.classes()).toContain('pointer-events-none')
    expect(wrapper.find('button').element.tabIndex).toBe(-1)

    await wrapper.trigger('click')
    expect(wrapper.emitted('click')).toBeUndefined()
  })
})
