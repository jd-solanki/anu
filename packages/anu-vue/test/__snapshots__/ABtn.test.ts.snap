// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing ABtn > can mount component 1`] = `
"<button tabindex=\\"0\\" class=\\"inline-flex whitespace-nowrap justify-center items-center relative a-btn states a-title-opacity-100 a-subtitle-opacity-100 [--a-layer-opacity:1] text-white\\" style=\\"--a-title-c:0, 0%, 100%; --a-subtitle-c:0, 0%, 100%; --a-layer-c:var(--a-primary); background: hsla(var(--a-layer-c),var(--a-layer-opacity));\\">
  <!-- ℹ️ Don't render spinner if not using loading -->
  <!--v-if-->
  <div class=\\"a-btn-content\\" data-no-reference=\\"\\">
    <!--v-if-->Click me
    <!--v-if-->
  </div>
</button>"
`;

exports[`Testing ABtn > has proper classes 1`] = `
[
  "inline-flex",
  "whitespace-nowrap",
  "justify-center",
  "items-center",
  "relative",
  "a-btn",
  "states",
  "a-title-opacity-100",
  "a-subtitle-opacity-100",
  "[--a-layer-opacity:1]",
  "text-white",
]
`;
